### PR TITLE
Feature 894 client parameter flexibility

### LINKED ISSUE #894: Client: more flexibility in passing parameters
# Currently
When using the client, users are forced to put the action (e.g. "scan") at last position.
Example:
`sechub -project project1 scan`
When moving the position of the action, you may get confusing error messages.

# Wanted
Enable putting the action at any position when calling the client
So e.g. this should be accepted also:
`sechub scan -project project1`

### PR DESCRIPTION
**Pull request by sven-dmlr**
- more flexibility in passing args (action can be passed other than last arg)
- default to help screen on empty action
- unit test added

closes #894

### CODE REVIEW

File: sechub-cli/src/mercedes-benz.com/sechub/cli/config_test.go [hunk 1]
```diff
@@ -133,3 +133,30 @@ func Example_validateRequestedReportFormatMakesLowercase2() {
 	// NOTICE: Converted requested report format 'Json' to lowercase. Because it contained uppercase characters, which are not accepted by SecHub server.
 	// json
 }
+
+func Example_normalizeCMDLineArgs() {
```
Comment (Jeeppler) [inline sechub-cli/src/mercedes-benz.com/sechub/cli/config_test.go:137]: You are testing only the happy path. How about the following cases:

~~~
argListXY := []string{"./sechub", "-jobUUID",  "getReport", "my-sechub.json", "-configfile", "3bdcc5c5-c2b6-4599-be84-f74380680808"}
argListYX := []string{"./sechub", "-jobUUID",  "my-sechub.json", "-configfile", "3bdcc5c5-c2b6-4599-be84-f74380680808", "getReport"}
argListXX := []string{"./sechub", "getReport", "-jobUUID",  "my-sechub.json", "-configfile", "3bdcc5c5-c2b6-4599-be84-f74380680808"}
~~~

**Review by Jeeppler — CHANGES_REQUESTED** [review]
It is not clear to me how the CLI will behave in edge cases, where values are messed up. Please have a look at the comment.

File: sechub-cli/src/mercedes-benz.com/sechub/cli/config_test.go (hunk 1 above)
Comment (Jeeppler) [reply sechub-cli/src/mercedes-benz.com/sechub/cli/config_test.go:137]: Okay.

**Comment by sven-dmlr** [comment]
@Jeeppler thanks for reviewing :+1:

**Pull request merged**